dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/PolicyExecutorImpl.java

```diff
@@ -679,12 +680,37 @@ void expediteGlobal(PollingTask pollingTask) {
             if (task == null)
                 throw new NullPointerException();
 
-        // TODO replace temporary code with proper untimed implementation. This will allow some tests to be written in advance,
+        InvokeAnyCompletionTracker tracker = new InvokeAnyCompletionTracker(taskCount);
+        ArrayList<PolicyTaskFutureImpl<T>> futures = new ArrayList<PolicyTaskFutureImpl<T>>(taskCount);
         try {
-            return invokeAny(tasks, Long.MAX_VALUE, TimeUnit.NANOSECONDS);
-        } catch (TimeoutException x) {
-            throw new UnsupportedOperationException(x);
+            // submit all tasks
+            for (Callable<T> task : tasks) {
+                PolicyTaskFutureImpl<T> taskFuture = new PolicyTaskFutureImpl<T>(this, task, tracker);
+
+                // check if done before enqueuing more tasks
+                if (tracker.hasSuccessfulResult())
+                    break;
+
```

Comment:
Do we want to exit out of the loop here if a RejectedExecutionException is thrown because the queue is full?  Or should we catch that exception (to possibly throw later?) and then try to enqueue the rest of the tasks

Reply:
RejectedExecutionException can happen in a couple of cases here.  First, if the executor is shutdown, in which case we would never want to retry for another task, because the shutdown state is not reversible.  Second, if the submit is interrupted, in which case we want to return immediately (with InterruptedException) to honor the interrupt.  Third, if, when attempting to submit one of the tasks, the queue is full and we have exceeded the maxWaitForEnqueue the task submission can abort per the Abort/runIfQueueFull=false policy.  If one task was prevented from enqueuing for this reason, then for any of the other tasks there would have been the same failure to enqueue.  Essentially, this is asking whether we should allow some tasks to be rejected and not be submitted while we continue with attempts to submit the subsequent tasks that were supplied to invokeAny.  We had to address this for untimed invokeAll, where the JavaDoc was very clear in specifying that RejectedExecutionException be raised "if any task cannot be scheduled for execution".  The JavaDoc for untimed invokeAny is less precise here. It specifies raising RejectedExecutionException, "if tasks cannot be scheduled for execution".  It does not state "all tasks", nor does it state "any task", and so could be interpreted as either or anything in between.  That said, I would favor aligning with the behavior for invokeAll, which also seems the most desirable here.  The maxWaitForEnqueue is intended to provide some buffer for the possibility that the queue might not be cleared out fast enough (by submitted tasks being assigned to available threads) for submits to be accepted instantaneously.  If maxWaitForEnqueue wasn't able to address that in the case of the first task, I don't see a lot of value in allowing the subsequent tasks in the group to serve as retries on the maxWaitForEnqueue.  The error should be surfaced right away on the first time that a task submit couldn't be accepted within the interval of the configured maxWaitForEnqueue to give the user the opportunity to decide how to respond.  Let me know if you have a different view on this.